src/vm/ceeload.cpp

```diff
@@ -528,7 +528,7 @@ void Module::Initialize(AllocMemTracker *pamTracker, LPCWSTR szName)
     m_dwTransientFlags &= ~((DWORD)CLASSES_FREED);  // Set flag indicating LookupMaps are now in a consistent and destructable state
 
 #ifdef FEATURE_READYTORUN
-    if (!HasNativeImage() && !IsResource())
+    if (!HasNativeImage() && !IsResource() && ReadyToRunInfo::IsReadyToRunEnabled())
```

Comment:
We have the same check in `ReadyToRunInfo::Initialize`. We should either remove it from there, or reorder the checks in `ReadyToRunInfo::Initialize` and not have it here.

Reply:
Ok, I've missed that. I'll remove the check here and reorder the checks in the ReadyToRunInfo::Initialize.